packages/tabs/src/TabTitle.tsx

```diff
@@ -125,6 +131,23 @@ const textOverflowStyles = css`
   text-overflow: ellipsis;
 `;
 
+function useDocumentActiveElement() {
+  const [activeEl, setActiveEl] = useState<Element | null>(null);
+
+  const handleFocusIn = useCallback(() => {
+    setActiveEl(document.activeElement);
+  }, []);
+
+  useEffect(() => {
+    document.addEventListener('focusin', handleFocusIn);
```

Comment:
This looks a lot like we're adding a new event listener for every single TabTitle on the page. Even though we clean these up, that's eating up more memory than we need to.

Reply:
Removed